discussion/src/main/java/org/openedx/discussion/presentation/comments/DiscussionCommentsViewModel.kt

```diff
@@ -292,7 +297,11 @@ class DiscussionCommentsViewModel(
     fun createComment(rawBody: String) {
         viewModelScope.launch {
             try {
-                val response = interactor.createComment(thread.id, rawBody, null)
+                val isCaptchaEnabled = config.getRecaptchaConfig().isEnabled &&
```

Comment:
instead of checking captchaFlag in every VM and then get the token, we can move this logic in repository.
```
    suspend fun getRecaptchaToken(courseId: String, recaptchaAction: RecaptchaAction): String {
        val isCaptchaEnabled = config.getRecaptchaConfig().isEnabled &&
                getCourseDiscussionConfig(courseId, false).isCaptchaEnabled
        return if (isCaptchaEnabled) recaptchaManager.getActionToken(recaptchaAction) else ""
    }

Reply:
it will reduce the VM workload and also the extra work & injection in VM and test classes